endpoints/materials.py

```diff
@@ -1,116 +1,88 @@
-
-        Returns:
-            list[dict]
-        """
-        return self.request('GET', self.name, params=params, headers=self.headers)
-
-    def get_material(self, mid, params=None):
-        """
-        Returns a material with a given ID.
-
-        Args:
-            mid (str): material ID.
-            params (dict): a dictionary of parameters passed to materials endpoint.
-                includeCharacteristics (bool): whether to include material's characteristics.
-
-        Returns:
-             dict: material.
-        """
-        return self.request('GET', '/'.join((self.name, mid)), params=params, headers=self.headers)
-
-    def get_materials_by_formula(self, formula, params=None):
-        """
-        Returns a list of materials with a given formula.
-
-        Args:
-            formula (str): material's formula.
-            params (dict): a dictionary of parameters passed to materials endpoint.
-                pageSize (int): page size. Defaults to 20.
-                pageIndex (int): page index to return. Defaults to 0.
-                includeCharacteristics (bool): whether to include material's characteristics.
-
-        Returns:
-            list[dict]
-        """
-        query = {"query": {"formula": formula}}
-        params = params.update(query) if params else query
-        return self.request('GET', self.name, params=params, headers=self.headers)
-
-    def delete_material(self, mid):
-        """
-        Deletes a given material.
-
-        Args:
-            mid (str): material ID.
-        """
-        return self.request('DELETE', '/'.join((self.name, mid)), headers=self.headers)
 
-    def update_material(self, mid, kwargs):
+    def import_from_file(self, name, content, owner_id=None, format="poscar", tags=()):
         """
-        Updates a material with given key-values in kwargs.
+        Imports a material from the given file.
 
         Args:
-            mid (str): material ID.
-            kwargs (dict): a dictionary of key-values to update.
+            name (str): material name.
+            content (str): material as string.
+            format (str): material format, either cif or poscar.
+            owner_id (str): owner ID. Material is created under user's default account by default.
+            tags (tuple[str]) a list of tags that should be assigned to the material.
 
         Returns:
-             dict: updated material.
+            dict
         """
-        headers = dict([('Content-Type', 'application/json')])
-        headers.update(self.headers)
-        return self.request('PATCH', '/'.join((self.name, mid)), data=json.dumps(kwargs), headers=headers)
+        data = {"name": name, "content": content, "format": format, "ownerId": owner_id, "tags": tags}
+        return self.request('POST', "/".join((self.name, "import")), headers=self.headers, data=json.dumps(data))
 
-    def create_material(self, material):
+    def _get_materialsproject_url(self, material_id):
```

Comment:
Any external functionality should be isolated. In utils, perhaps.

Reply:
Done.